app/uk/gov/hmrc/transitmovementsrouter/controllers/ObjectStoreURIExtractor.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.controllers
+
+import cats.data.EitherT
+import play.api.mvc.BaseController
+import uk.gov.hmrc.transitmovementsrouter.controllers.ObjectStoreURIExtractor.expectedUriPattern
+import uk.gov.hmrc.transitmovementsrouter.controllers.errors.PresentationError
+import uk.gov.hmrc.transitmovementsrouter.models.ObjectStoreResourceLocation
+import uk.gov.hmrc.transitmovementsrouter.models.ObjectStoreURI
+
+import scala.concurrent.Future
+import scala.util.matching.Regex
+
+object ObjectStoreURIExtractor {
+
+  // The URI consists of the service name in the first part of the path, followed
+  // by the location of the object in the context of that service. As this service
+  // targets common-transit-convention-traders' objects exclusively, we ensure
+  // the URI is targeting that context. This regex ensures that this is the case.
+  val expectedUriPattern: Regex = "^common-transit-convention-traders/(.+)$".r
+}
+
+trait ObjectStoreURIExtractor {
+  self: BaseController =>
+
+  def extractObjectStoreResourceLocation(objectStoreURI: ObjectStoreURI): EitherT[Future, PresentationError, ObjectStoreResourceLocation] =
+    EitherT(
+      Future.successful(
+        for {
```

Comment:
```
    override val asUri: String =
      s"${directory.asUri}${URLEncoder.encode(fileName.stripSuffix("/"), "UTF-8")}"
  }
```

This takes the file directory and name, so it won't be common-transit-convention-traders? In anycase, if you want to extract from it you can just do objectSummary.location.directory or something like that

Reply:
the uri is going to be ``` common-transit-convention-traders/movements/movementID/x-conversation—id/formattedDateTime.xml ```  (I have tested this with poc) . so from this we are removing  common-transit-convention-traders/  using above pattern as with ``` getObject ``` Object store method we just need to pass movements/movementID/x-conversation—id/formattedDateTime.xml the same things that we have passed from putObject Object store uri method (while uploading file from upscan to object store)